homeassistant/helpers/discovery.py

```diff
@@ -129,7 +129,7 @@ def load_platform(
     hass: core.HomeAssistant,
     component: Platform | str,
     platform: str,
-    discovered: DiscoveryInfoType,
+    discovered: DiscoveryInfoType | None,
```

Comment:
This seems incorrect, as `async_load_platform` doesn't allow for it either.

Reply:
I'm not sure what you mean. I have adjusted `async_load_platform` also...